tests/PCA.py

```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+import numpy as np
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+import matplotlib
+import matplotlib.pyplot as plt
+
+#defining the function for PCA
+def Compute_PCA(data_to_compute, num_components, show_results=False):
+    """THIS FUNCTOIN TAKES DATAFRAME AS INPUT, CALCULATES THE PRINCIPLE COMPONENTS AND RETURS A DATAFRAME CONTAINING THE PRINCIPAL COMPONETS INSIDE IT"""
+    try:
```

Comment:
We want to avoid using generic try - except -blocks. If there are some specific requirements for the data, write those as checks (using if expressions for example) and raise a custom expression if something is wrong. So please remove the try and except and if needed, add some other code to handle error situations. Again I suggest taking look at other toolkit functions.